contracts/zero-ex/test/features/multiplex_test.ts

```diff
@@ -0,0 +1,771 @@
+            const multiHopFillData = {
+                tokens: [ETH_TOKEN_ADDRESS, WETH_ADDRESS, USDC_ADDRESS, DAI_ADDRESS],
+                sellAmount,
+                calls: [wrapEthCall, uniswapCall, curveFqtCall],
+            };
+            const tx = await multiplex
+                .multiHopFill(multiHopFillData, constants.ZERO_AMOUNT)
+                .awaitTransactionSuccessAsync(
+                    { from: rfqMaker, gasPrice: 0, value: sellAmount },
+                    { shouldValidate: false },
+                );
+            console.log(`${tx.gasUsed} gas used`);
+
+            const [wethDepositEvent] = filterLogsToArguments<WETH9DepositEventArgs>(tx.logs, WETH9Events.Deposit);
+            expect(wethDepositEvent._owner, 'WETH Deposit event _owner').to.equal(zeroEx.address);
+            expect(wethDepositEvent._value, 'WETH Deposit event _value').to.bignumber.equal(sellAmount);
+
+            const [uniswapEvent] = filterLogsToArguments<IUniswapV2PairSwapEventArgs>(
+                tx.logs,
+                IUniswapV2PairEvents.Swap,
+            );
+            expect(uniswapEvent.sender, 'Uniswap Swap event sender').to.equal(zeroEx.address);
+            expect(uniswapEvent.to, 'Uniswap Swap event to').to.equal(flashWalletAddress);
+            const uniswapInputAmount = BigNumber.max(uniswapEvent.amount0In, uniswapEvent.amount1In);
+            expect(uniswapInputAmount, 'Uniswap Swap event input amount').to.bignumber.equal(sellAmount);
+            const uniswapOutputAmount = BigNumber.max(uniswapEvent.amount0Out, uniswapEvent.amount1Out);
+            expect(uniswapOutputAmount, 'Uniswap Swap event output amount').to.bignumber.gt(0);
+
+            const [bridgeFillEvent] = filterLogsToArguments<BridgeAdapterBridgeFillEventArgs>(
+                tx.logs,
+                BridgeAdapterEvents.BridgeFill,
+            );
+            expect(bridgeFillEvent.source).to.bignumber.equal(BridgeSource.Curve);
+            expect(bridgeFillEvent.inputToken).to.equal(USDC_ADDRESS);
+            expect(bridgeFillEvent.outputToken).to.equal(DAI_ADDRESS);
+            expect(bridgeFillEvent.inputTokenAmount).to.bignumber.equal(uniswapOutputAmount);
+            expect(bridgeFillEvent.outputTokenAmount).to.bignumber.gt(0);
+        });
+        it('MultiplexFeature.multiHopFill() complex scenario', async () => {
+            /*
+
+                                      /––––PLP–––> USDC
+                                     /                \
+                                    /                 PLP
+                                   /––Uni (via USDC)–––\
+                                  /                    V
+                ETH ––wrap––> WETH ––––Uni/Sushi–––> USDT ––Sushi––> LON
+                                \                                      ^
+                                 ––––––––––––––– Uni –––––––––––––––––/
+            */
+            // Taker has to have approved the EP for the intermediate tokens :/
```

Comment:
We could avoid this if we allow the EP itself to hold the intermediate tokens, but this may cause some wonky behavior... alternatively, we can write a Transformer that calls the Multiplex feature, at the cost of some efficiency